### PR TITLE
Feature/task 3

### PR DESCRIPTION
**Pull request by JarekParal**

### CODE REVIEW

File: src/reducers/itemsReducer.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const doga = new ItemRecord({
+    id: idGenerator1(),
+    text: 'Doga'
+  });
+
+  const defaultState = initialState.set(dog.id, dog);
+
+  let state;
+
+  it('should return the initial state', () => {
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/reducers/itemsReducer.test.js]: Every time when you run the test?

[the remark this answers is not in the document]
Comment (JarekParal) [reply src/reducers/itemsReducer.test.js]: Ok, now I understand ;-).

File: src/reducers/itemsReducer.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { OrderedMap } from 'immutable';
+import {
+  ITEM_CREATED,
+  ITEM_EDITED,
+  ITEM_DELETED,
+} from '../actions/actionTypes';
+import { ItemRecord } from '../models/ItemRecord';
+
+export const initialState = new OrderedMap({});
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/reducers/itemsReducer.js]: "you create the OrderedMap without the new keyword"

**Comment by JarekParal** [comment]
Finish

File: src/reducers/item.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export const item = (state, action) => {
+  switch (action.type) {
+    case EDIT_ITEM:
+      return (
+        state
+          .setIn([action.payload.id, 'text'], action.payload.text)
+      );
+
+    case DELETE_ITEM:
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/reducers/item.js]: Yes, that is true. I didn't think about that when I moved this probably :).

File: src/reducers/items.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        state
+          .set(action.payload.id,
+            new ItemRecord({
+              id: action.payload.id,
+              text: action.payload.text,
+            }))
+      );
+
+    default:
+      return item(state, action);
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/reducers/items.js]: Do you think something like this?
```js
export const items = (...) {
  switch (action.type) {
    case CREATE_ITEM:
       ...
    case EDIT_ITEM:
      return (item(state, action));
    case DELETE_ITEM:
      return (item(state, action));
    default:
      return state;
}
```
I was thinking about this implementation but then I added some functionality of `item` to `items`. And what about when `item` has 10 cases?

**Comment by JarekParal** [comment]
Finish

**Comment by matus12** [comment]
I have gone through the changes, discussed few small issues in person, @vaclavm will look at the code before closing the pull request.

**Pull request merged**